### PR TITLE
Onboarding new tests

### PR DESCRIPTION
**Pull request by burtonemily**
Adds in new tests
- Disable disappearing messages in 1:1 conversations
- Delete contact (conversation)
- Hide recovery password permanently

Removes type checking on strings returned from JSON file for locales
Adds in ENUM for Username instead of type
Starts to implement modal string checks

### CODE REVIEW

File: run/test/specs/locators/disappearing_messages.ts
Comment (Bilb) [inline run/test/specs/locators/disappearing_messages.ts]: when the android and the ios are the same, you can just write
```ts
export class DisappearingMessagesSubtitle extends LocatorsInterface {
  public build(): StrategyExtractionObj {
    switch (this.platform) {
      case 'android':
      case 'ios':
        return {
          strategy: 'accessibility id',
          selector: `Disappearing messages type and time`,
        };
    }
  }
}
```

and both will share the same values.
(there are a few duplicates in this file, check the others too!)

File: run/test/specs/locators/index.ts
Comment (Bilb) [inline run/test/specs/locators/index.ts]: same comment

File: run/test/specs/locators/settings.ts
Comment (Bilb) [inline run/test/specs/locators/settings.ts]: same comment

File: sessionReporter.ts
Comment (Bilb) [inline sessionReporter.ts]: can be removed altogether?

File: run/types/sessionIt.ts
Comment (Bilb) [inline run/types/sessionIt.ts]: can be removed altogether?

File: run/types/sessionIt.ts
Comment (Bilb) [inline run/types/sessionIt.ts]: you should be able to get rid of that  `// eslint-disable-next-line @typescript-eslint/require-await` line by removing the `async` here `    test.skip(testName, async () => {`

File: run/types/sessionIt.ts
Comment (Bilb) [inline run/types/sessionIt.ts]: same comment

File: run/test/specs/disappear_after_send_off_1o1.spec.ts
Comment (Bilb) [inline run/test/specs/disappear_after_send_off_1o1.spec.ts]: could be named alice and bob :D

File: run/test/specs/linked_device_profile_picture_syncs.spec.ts
Comment (Bilb) [inline run/test/specs/linked_device_profile_picture_syncs.spec.ts]:
```ts

if (actualPixelColorLinked !== expectedPixelHexColour) {
 throw new Error(
    `Device1: Colour isn't ${expectedPixelHexColour}, it is: ` + actualPixelColorLinked
  );
} 
  console.log('Device 2: Colour is correct on linked device');

  ```

File: run/test/specs/linked_device_profile_picture_syncs.spec.ts
Comment (Bilb) [inline run/test/specs/linked_device_profile_picture_syncs.spec.ts]: could be `new UserSettings(device2)` (and in a bunch of other places too). I guess you udpate each tests one after the other?

File: run/test/specs/utils/disappearing_control_messages.ts
Comment (Bilb) [inline run/test/specs/utils/disappearing_control_messages.ts]: not a reason to make the test fail?  I mean this is expected?

File: run/types/DeviceWrapper.ts
Comment (Bilb) [inline run/types/DeviceWrapper.ts:662]: I should have a look at this

File: run/test/specs/user_actions_delete_contact.spec.ts
Comment (Bilb) [inline run/test/specs/user_actions_delete_contact.spec.ts]: you should be able to await Promise.all those 2 calls

File: run/test/specs/user_actions_delete_contact.spec.ts
Comment (Bilb) [inline run/test/specs/user_actions_delete_contact.spec.ts]: you shouldn't need a platform argument on navigateBack, as it is a class method. You know the platform from the class instance

File: run/test/specs/user_actions_delete_contact.spec.ts
Comment (Bilb) [inline run/test/specs/user_actions_delete_contact.spec.ts:42]: are those 2 the same? Should the one commented be removed?

File: run/types/DeviceWrapper.ts
Comment (Bilb) [inline run/types/DeviceWrapper.ts]: the name is misleading, maybe rename it to `removeNewLines`
also, isn't playwright supposed to do those checks for you?

File: run/test/specs/disappear_after_send_off_1o1.spec.ts
Comment (Bilb) [inline run/test/specs/disappear_after_send_off_1o1.spec.ts:90]: you don't want to allow the types not built in doesElementExist?

File: .eslintrc.js
Comment (Bilb) [inline .eslintrc.js:18]: this might cause issues down the line, but why not

**Review by Bilb — CHANGES_REQUESTED** [review]

File: run/test/specs/locators/disappearing_messages.ts
Comment (burtonemily) [reply run/test/specs/locators/disappearing_messages.ts]: Fixed

File: run/test/specs/linked_device_profile_picture_syncs.spec.ts
Comment (burtonemily) [reply run/test/specs/linked_device_profile_picture_syncs.spec.ts]: Updated all instances

File: run/test/specs/utils/disappearing_control_messages.ts
Comment (burtonemily) [reply run/test/specs/utils/disappearing_control_messages.ts]: Forgot to remove this once the bug was reported, will remove now

File: run/types/DeviceWrapper.ts
Comment (burtonemily) [reply run/types/DeviceWrapper.ts:662]: As in you're not happy with this?

File: run/test/specs/user_actions_delete_contact.spec.ts
Comment (burtonemily) [reply run/test/specs/user_actions_delete_contact.spec.ts]: Removed

File: run/test/specs/user_actions_delete_contact.spec.ts
Comment (burtonemily) [reply run/test/specs/user_actions_delete_contact.spec.ts:42]: No the top line that is commented out is the function that checks if the modal strings are correct but in this PR we didn't have a way to add args into the modal checks, so i have just commented it out for now. Will add a TODO to uncommented once the fix i've implemented is merged. 
The second line is just clicking the button

File: run/types/DeviceWrapper.ts
Comment (burtonemily) [reply run/types/DeviceWrapper.ts]: Playwright currently does not do these checks, so had to create my own function. Will rename.

File: run/test/specs/disappear_after_send_off_1o1.spec.ts
Comment (burtonemily) [reply run/test/specs/disappear_after_send_off_1o1.spec.ts:90]: This was to allow a custom maxWait. The current implementation doesn't allow the new locator type with any extra variables. I tried a bunch of different ways and this was the best I could come up with

File: .eslintrc.js
Comment (burtonemily) [reply .eslintrc.js:18]: This is something Ryan added, so I have no idea what it is..

File: run/test/specs/community_tests_image.spec.ts
Comment (Bilb) [inline run/test/specs/community_tests_image.spec.ts]: to remove?

File: run/test/specs/group_disappearing_messages_link.spec.ts
Comment (Bilb) [inline run/test/specs/group_disappearing_messages_link.spec.ts]: not sure why you keep all of those commented navigateBack

File: run/types/DeviceWrapper.ts
Comment (Bilb) [inline run/types/DeviceWrapper.ts]: if statements could be swap to have the throw and then normal execution

**Review by Bilb — APPROVED** [review]

File: run/types/DeviceWrapper.ts
Comment (Bilb) [reply run/types/DeviceWrapper.ts:662]: not really no. Will check it out once it's merged

File: run/test/specs/group_disappearing_messages_link.spec.ts
Comment (burtonemily) [reply run/test/specs/group_disappearing_messages_link.spec.ts]: For fun

**Review by Bilb — APPROVED** [review]

**Pull request merged**
